src/components/transportation-icon/index.tsx

```diff
@@ -1,40 +1,47 @@
 import React from 'react';
-import {View, StyleProp, ViewStyle} from 'react-native';
+import {View, StyleProp, ViewStyle, AccessibilityProps} from 'react-native';
 import {
   BusSide,
   TramSide,
   TrainSide,
   PlaneSide,
   FerrySide,
+  WalkingPerson,
 } from '../../assets/svg/icons/transportation';
 import {LegMode} from '../../sdk';
 import {StyleSheet} from '../../theme';
 import {SvgProps} from 'react-native-svg';
 import transportationColor from '../../utils/transportation-color';
+import {useTranslation} from '../../translations';
+import {getTranslatedModeName} from '../../utils/transportation-names';
 
 export type TransportationIconProps = {
   mode?: LegMode;
   publicCode?: string;
   style?: StyleProp<ViewStyle>;
-  circleStyle?: StyleProp<ViewStyle>;
 };
 
 const TransportationIcon: React.FC<TransportationIconProps> = ({
   mode,
   publicCode,
   style,
-  circleStyle,
   children,
 }) => {
   const styles = useStyle();
-  const {fill, icon} = transportationColor(mode, publicCode);
+  const {t} = useTranslation();
+  const {color} = transportationColor(mode, publicCode);
 
   return (
-    <View style={[styles.circle, {backgroundColor: fill}, circleStyle]}>
+    <View style={styles.iconContainer}>
```

Comment:
Note to self: I've also had to do some changes here, but seems as I can just throw them away. (I made an additional component to avoid having to rewrite and break journey details)